sites/bestbuy.py

```diff
@@ -6,327 +6,554 @@
+                        self.order_id), json={}, headers=headers,
+                    verify=False)
+                r = json.loads(r.text)
+                if r['id'] == self.order_id:
+                    self.status_signal.emit(
+                        {"msg": "Payment Refreshed",
+                         "status": "normal"})
+                    return
             except Exception as e:
-                self.status_signal.emit({"msg":"Error Submitting Payment (line {} {} {})".format(sys.exc_info()[-1].tb_lineno, type(e).__name__, e),"status":"error"})
+                self.status_signal.emit({
+                    "msg": "Error Submitting Order (line {} {} {})".format(
+                        sys.exc_info()[-1].tb_lineno, type(e).__name__, e),
+                    "status": "error"})
                 time.sleep(self.error_delay)
-    
+
     def submit_order(self):
         headers = {
-            "accept": "application/com.bestbuy.order+json",
-            "accept-encoding": "gzip, deflate, br",
-            "accept-language": "en-US,en;q=0.9,zh-CN;q=0.8,zh;q=0.7",
-            "content-type": "application/json",
-            "origin": "https://www.bestbuy.com",
-            "referer": "https://www.bestbuy.com/checkout/r/payment",
-            "user-agent": settings.userAgent,
-            "x-user-interface": "DotCom-Optimized"
+            "Accept": "application/com.bestbuy.order+json",
+            "Accept-Encoding": "gzip, deflate",
+            "Accept-Language": "en-US,en;q=0.9,zh-CN;q=0.8,zh;q=0.7",
+            "Content-Type": "application/json",
+            "Origin": "https://www.bestbuy.com",
+            "Referer": "https://www.bestbuy.com/checkout/r/payment",
+            "User-Agent": settings.userAgent,
+            "X-User-Interface": "DotCom-Optimized"
+        }
+        body = {
+            "browserInfo": {
+                "javaEnabled": False,
+                "language": "en-US",
+                "userAgent": settings.userAgent,
+                "height": "1057",
+                "width": "1057",
+                "timeZone": "240",
+                "colorDepth": "24"
+            }
         }
-        body = {"browserInfo":{"javaEnabled":False,"language":"en-US","userAgent":settings.userAgent,"height":"1057","width":"1057","timeZone":"240","colorDepth":"24"}}
         while True:
-            self.status_signal.emit({"msg":"Submitting Order","status":"alt"})
+            self.status_signal.emit({"msg": "Submitting Order", "status": "alt"})
             try:
-                r = self.session.post("https://www.bestbuy.com/checkout/orders/{}/".format(self.order_id),json=body,headers=headers)
+                r = self.session.post("https://www.bestbuy.com/checkout/orders/{}/".format(
+                    self.order_id), json=body, headers=headers, verify=False)
                 r = json.loads(r.text)
                 try:
                     r = r["errors"][0]
```

Comment:
Isnt the second half of the inner if statement dead code? You have it after a return statement.

```
       if errors[0].get("errorCode") == "PAY_SECURE_REDIRECT":
            self.status_signal.emit(
                {"msg": "3DSecure Found, Starting Auth Process", "status": "error"}
            )
            return False, errors[0]["paySecureResponse"]["stepUpJwt"]
            self.status_signal.emit(
                {"msg": "3DSecure Found, Starting Auth Process", "status": "error"}
            )
            self.status_signal.emit(
                {"msg": f"Payment error {errors[0]['errorCode']}", "status": "error"}
            )
            return False, errors[0]["paySecureResponse"]["stepUpJwt"]
```

Reply:
> Isnt the second half of the inner if statement dead code? You have it after a return statement.
> 
> ```
>        if errors[0].get("errorCode") == "PAY_SECURE_REDIRECT":
>             self.status_signal.emit(
>                 {"msg": "3DSecure Found, Starting Auth Process", "status": "error"}
>             )
>             return False, errors[0]["paySecureResponse"]["stepUpJwt"]
>             self.status_signal.emit(
>                 {"msg": "3DSecure Found, Starting Auth Process", "status": "error"}
>             )
>             self.status_signal.emit(
>                 {"msg": f"Payment error {errors[0]['errorCode']}", "status": "error"}
>             )
>             return False, errors[0]["paySecureResponse"]["stepUpJwt"]
> ```

pretty sure that's a mistake since the lines 2 and 3 of the code block is also duplicated. But we understand his point overall.